build/tools/Automation/DotNet/DotNet.cs

```diff
@@ -33,148 +31,142 @@ namespace Microsoft.Oryx.Automation
     ///           Oryx tests.
     ///     - Updating versionsToBuild.txt
     /// </Summary>
-    public class DotNet : Program
+    public class DotNet
     {
-        private string repoAbsolutePath = string.Empty;
-        private HashSet<string> prodSdkVersions = new HashSet<string>();
+        private readonly IHttpClient httpClient;
+        private readonly ILogger logger;
+        private readonly IVersionService versionService;
+        private readonly IYamlFileReaderService yamlFileReaderService;
 
-        public DotNet(string repoAbsolutePath, HashSet<string> prodSdkVersions)
+        public DotNet(IHttpClient httpClient, ILogger logger, IVersionService versionService, IYamlFileReaderService yamlFileReaderService)
         {
-            this.repoAbsolutePath = repoAbsolutePath;
-            this.prodSdkVersions = prodSdkVersions;
+            this.httpClient = httpClient;
+            this.logger = logger;
+            this.versionService = versionService;
+            this.yamlFileReaderService = yamlFileReaderService;
         }
 
-        /// <Summary>
-        /// Gets DotNet's new release version and sha.
-        ///
-        /// This is accomplished by:
-        ///     - Checking release meta data url if there's a new release
-        ///     - If new release, then store release information into PlatformConstants
-        ///        Otherwise don't store anything
-        /// </Summary>
-        /// <param name="dateTarget">yyyy-mm-dd format string that defaults to today's date</param>
-        /// <returns>PlatformConstants used later to update constants.yaml</returns>
-        public override async Task<List<PlatformConstant>> GetPlatformConstantsAsync(string dateTarget)
+        public async Task RunAsync(string oryxRootPath)
```

Comment:
We should make `oryxRootPath` a private variable in this class that we set after it's been provided to `RunAsync` so that every method is able to easily reference it without us having to pass it around in each method. This value should only ever be provided once on application startup (since it's an argument to the app), so there shouldn't be any concern about having it overridden or changed during the lifespan of it executing.

Reply:
Updated to use via RunAsync